src/coreclr/nativeaot/System.Private.CoreLib/src/System/Threading/ManagedThreadId.cs

```diff
@@ -246,11 +248,16 @@ public static void RecycleId(int id)
             }
         }
 
+        // We do -1 adjustment so that uninitialized result would be -1 and
```

Comment:
The `-1` trick can be local to the few places in ObjectHeader.cs that need it, like: `if ((uint)(currentThreadID - 1) < (uint)SBLK_MASK_LOCK_THREADID)`. It does not need to be spread over multiple files like this.

Reply:
Hmm, that would add extra condition though. Otherwise Release can't tell if lock is owned by nobody or by a thread with uninitialized thread ID.

Reply:
When uninitialized ID is -1, it can't own a thin lock, so we do not need to check if the ID is initialized. 
0 will match with unowned state.

Reply:
ok, it can stay as is then.

Reply:
I think Release can apply `currentThreadID |= (currentThreadID - 1) >> 31;`  and have the same effect as if default was -1
If we store ID in off-by-one form, we still need to do "-1", so this would be just 2 extra operations and the default can be 0.

The Acquire can do `if ((uint)(currentThreadID - 1) < (uint)SBLK_MASK_LOCK_THREADID)` as you suggested.